thirdparty/faiss/faiss/impl/index_read.cpp

```diff
@@ -234,6 +236,9 @@ InvertedLists* read_InvertedLists(IOReader* f, int io_flags) {
                     size_t seg_off = lca->get_segment_offset(i, j);
                     READANDCHECK(lca->codes[i][j].data_.data(), seg_size * lca->code_size);
                     READANDCHECK(lca->ids[i][j].data_.data(), seg_size);
+                    if (saveNormal) {
```

Comment:
"save_normals" looks redundant, you can use "if (lca->code_normals != nullptr)"

Reply:
'save_normals' represents the invertedLists should hold the normals value, instead of depending on the caller. If the 'save_normals' set, normals value must be provided，or else error will occur.